sklearn/model_selection/_validation.py

```diff
@@ -231,6 +287,10 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
     X_train, y_train = _safe_split(estimator, X, y, train)
     X_test, y_test = _safe_split(estimator, X, y, test, train)
 
+    n_scorers = len(scorers.keys())
+    test_scores = dict(zip(scorers.keys(), [error_score, ] * n_scorers))
```

Comment:
I mean you're not modifying this dict in place below. It's either all `error_score` or all retrieved from `_score`. Why not set it to `error_score` in the except handler?